packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -5,55 +5,90 @@
 
 import * as vscode from 'vscode'
 import * as fs from 'fs'
+import * as path from 'path'
 import { getLogger, Logger } from '../../shared/logger'
 import { localize } from '../../shared/utilities/vsCodeUtils'
 import { VueWebview } from '../../webviews/main'
 import { ExtContext } from '../../shared/extensions'
-//import { telemetry } from '../../shared/telemetry/telemetry'
+// TODO: Implement Telemetry ... import { telemetry } from '../../shared/telemetry/telemetry'
 import { AccessAnalyzer, S3 } from 'aws-sdk'
-import { exec } from 'child_process'
+import { execSync } from 'child_process'
+import { ToolkitError } from '../../shared/errors'
 const AmazonS3URI = require('amazon-s3-uri')
 
+const defaultTerraformConfigPath = path.join(__dirname, '..', 'terraformConfig', 'default.yaml')
+const tempFilePath = path.join(__dirname, 'policyDocument.json')
+const customPolicyCheckDiagnosticCollection = vscode.languages.createDiagnosticCollection('customPolicyCheck')
+const validatePolicyDiagnosticCollection = vscode.languages.createDiagnosticCollection('validatePolicy')
+
 export interface IamPolicyChecksInitialData {
     referenceFilePath: string
```

Comment:
Q: Why do we not have fields related to the CheckNoNewAccess here?

Reply:
I see that we're reusing these member variables for other check, I don't think we should be doing that. We should initialize the inputs for other API separately for `actions` in CheckNoNewAccess

Reply:
CheckNoNewAccess and CheckAccessNotGranted both go under the same diagnostic collection, as they share the same view in the UI.

Reply:
Why can't we have a separate diagnostic collection? What happens when new APIs are added to the custom policy checks

Reply:
We can have a separate diagnostic collection, but since they share the same view it may be difficult for users to determine which problems belong to which check. I imagine that users may execute the wrong checks when trying to resolve a different issue altogether. That will cause confusion as to why X warning will not go away, even though they are "running" the check again.

They will reuse the same collection.

Reply:
As discussed offline, we will share the diagnostic collection for now. Let's rename the variables and add documentation (through comments) to indicate that they are being shared across the checks